zerver/lib/display_recipient.py

```diff
@@ -122,101 +122,39 @@ def bulk_fetch_user_display_recipients(
     Returns dict mapping recipient_id to corresponding display_recipient
     """
 
-    if len(recipient_tuples) == 0:
-        return {}
+    get_recipient_id = lambda tup: tup[0]
+    get_type = lambda tup: tup[1]
 
-    recipient_id_to_type = {recipient[0]: recipient[1] for recipient in recipient_tuples}
+    personal_tuples = [tup for tup in recipient_tuples if get_type(tup) == Recipient.PERSONAL]
+    huddle_tuples = [tup for tup in recipient_tuples if get_type(tup) == Recipient.HUDDLE]
 
-    recipient_id_to_type_id = {recipient[0]: recipient[2] for recipient in recipient_tuples}
+    huddle_recipient_ids = [get_recipient_id(tup) for tup in huddle_tuples]
+    huddle_user_id_dict = bulk_get_huddle_user_ids(huddle_recipient_ids)
 
-    recipient_ids = [recipient[0] for recipient in recipient_tuples]
+    # Find all user ids whose UserProfiles we will need to fetch:
+    user_ids_to_fetch: Set[int] = set()
 
-    # Now we have to create display_recipients for personal and huddle messages.
-    # We do this via generic_bulk_cached_fetch, supplying appropriate functions to it.
+    for recipient_id, _, user_id in personal_tuples:
```

Comment:
replace `_` with `ignored_recipient_type` -- using `_` here as a dummy variable in Python can break i18n because `_` is used to tag translations too.

Reply:
Fixing